data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/HasTagsExpressionFunction.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.model.event.Event;
+import java.util.List;
+import java.util.Set;
+import javax.inject.Named;
+import java.util.function.Function;
+
+@Named
+public class HasTagsExpressionFunction implements ExpressionFunction {
+    public String getFunctionName() {
+        return "hasTags";
+    }
+
+    public Object evaluate(final List<Object> args, Event event, Function<Object, Object> convertLiteralType) {
+        if (args.size() == 0) {
+            throw new RuntimeException("hasTags() takes at least one argument");
+        }
+        Set<String> tags = event.getMetadata().getTags();
```

Comment:
I think we may benefit from adding the `addTags()` method to the `EventMetadata`.

This has a couple benefits in my opinion:
1. It separates the concerns - determining whether an Event has a tag versus interpreting the function arguments.
2. It creates a function that could end up being useful elsewhere.
3. You can make the tests for this focus more on the arguments.

```
boolean hasTags(List<String> tags) {
  for(String tag : tags) {
    if(!tags.contains(tag)) {
      return false;
    }
  }
  return true;
}
```

This code can then be something like:

```
if(!args.stream()
  .allMatch(a -> a instanceof String)) {
    throw new RuntimeException("...");
}
final List<String> tags = args.stream()
  .map(a -> (String) a)
  .collect(Collectors.toList());
return event.getMetadata().hasTags(tags);
```